Hangman Game/Player.py

```diff
@@ -0,0 +1,54 @@
+MAX_GUESSES = 6
+
+class Player: #Contains all the player details
+    
+    def __init__(self, name) -> None:
+        self._player_name = name
+        self.ResetPlayerDetails()
+    
+    def ResetPlayerDetails(self) -> None:
+        self._totalscore = 0
+        self._hintpanelty = 0
+        self._player_score = 0
+        self._hintTaken = 0
+        self._remaining_guess = MAX_GUESSES
+        self._present_valid_moves = list()
+    
+    def WrongGuess(self) -> None:
+        if self._remaining_guess>0 :
```

Comment:
Perhaps handle the case when the remaining guesses are already zero and raise an error. This will prevent programmer mistakes from being silently hidden.